### DISCUSSION TITLE
device-sdk-c

### DISCUSSION THREAD

**Discussion by wangqingpen** [edgexfoundry/community#259] [Q&A]
https://docs.edgexfoundry.org/2.1/getting-started/Ch-GettingStartedSDK-C/
![image](https://github.com/edgexfoundry/community/assets/72652818/f1bf3e43-a5ae-4fc4-aef5-f59152640560)
The following error is reported when the device-sdk-c is used:
CMake Error at /usr/share/cmake-3.13/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find CURL (missing: CURL_LIBRARY CURL_INCLUDE_DIR)
Call Stack (most recent call first):
  /usr/share/cmake-3.13/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake-3.13/Modules/FindCURL.cmake:74 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  CMakeLists.txt:24 (find_package)

-- Configuring incomplete, errors occurred!
See also "/home/edgex/device-sdk-c/build/release/CMakeFiles/CMakeOutput.log".

**Comment by jiekechoo** [comment]
> Could NOT find CURL (missing: CURL_LIBRARY CURL_INCLUDE_DIR)

You should install the `CURL` lib before build the device-c service.

**Comment by wangqingpen** [comment]
> > Could NOT find CURL (missing: CURL_LIBRARY CURL_INCLUDE_DIR)
> 
> You should install the `CURL` lib before build the device-c service.

I'm using the example you provided with the error code 0 , message: undefined
